csrc/include/natten/cuda/fna/kernel_forward.h

```diff
@@ -272,7 +280,11 @@ struct FusedNeighborhoodAttentionKernel {
       output_ptr += (first_query * o_strideM).sum() +
           (dilation_idx * o_stride_dilation).sum() + head_id * o_strideH;
 
-      if constexpr (kSupportsRPB) {
+      assert(
+        (!has_rpb || !kHasCausalDims) && "Causal NA does not support RPB yet.");
```

Comment:
Can you move this to host side instead?

I highly prefer moving assertions to outside the kernel as much as possible, I've had some bad experiences in the past with device side asserts.

We might already have a check for it on host, but if we don't, just add an if statement and raise an error in `kernel_forward.h`. There should be some checks in the already that you can copy paste.

Reply:
Done, please take a look if that's fine.